addons/csw/functions/fnc_initVehicle.sqf

```diff
@@ -63,14 +68,45 @@ if (_vehicle turretLocal [0]) then {
     };
 };
 
+if (_hasAutofireEnabled || _forceAutofireEnabled) then {
+    TRACE_3("hasAutofireEnabled",_vehicle,_hasAutofireEnabled,_forceAutofireEnabled);
+
+    private _mainTurret = configFile >> "CfgVehicles" >> _typeOf >> "turrets" >> "MainTurret";
+    private _weapon = (getArray (_mainTurret >> "weapons")) select 0;
+    private _weaponConfig = configFile >> "CfgWeapons" >> _weapon;
+    private _modifier = deg getNumber (_weaponConfig  >> "dispersion");
+
+    private _dispersionInModes = createHashMap;
+    private _modes = getArray (_weaponConfig >> "modes");
+    {
+        private _mode = _weaponConfig >> _x;
+        private _name = configName _mode;
+        private _dispersion = getNumber (_mode >> "artilleryDispersion");
+
+        _dispersionInModes set [_name, [_dispersion, _modifier]];
+    } forEach _modes;
+
+    private _animationSources = [
+        getText (_mainTurret >> "animationSourceBody"),
+        getText (_mainTurret >> "animationSourceGun")
+    ];
+
+    _vehicle setVariable [QGVAR(autofire_animations), _animationSources];
+    _vehicle setVariable [QGVAR(autofire_dispersion), _dispersionInModes];
+    _vehicle setVariable [QGVAR(autofire_defaultModes), createHashMap];
+    _vehicle setVariable [QGVAR(autofire), _forceAutofireEnabled];
+    _vehicle setVariable [QGVAR(autofire_force), _forceAutofireEnabled];
```

Comment:
Just in-case people don't want the forced part, adding this should give an easy scripted way to disable

```suggestion
    if (_vehicle isNil QGVAR(autofire_force)) then { _vehicle setVariable [QGVAR(autofire_force), _forceAutofireEnabled]; };
```